src/basic/math-util.h

```diff
@@ -0,0 +1,12 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include <math.h>
+
+#include "macro.h"
+
+/* On some optimization level, iszero(x) is converted to (x == 0.0), and emits warning -Wfloat-equal. */
+#define iszero_safe(x) (fpclassify(x) == FP_ZERO)
+
+/* To avoid x == y and triggering compile warning -Wfloat-equal. */
+#define fp_equal(x, y) iszero_safe(x - y)
```

Comment:
is this safe? i.e. what if x minus y actually leaves the defined range for double? Does the comparison still work?

at the very least x and y should be enclosed in `()`

Reply:
`x - y` is always in double (of course when at least one is double), but may be NaN or infinity. But in that case, this should returns false, and actually.

The missing `()` are added.